### DISCUSSION TITLE
Github actions populate the PR timeline so much, that it becomes difficult to read

### DISCUSSION THREAD

**Discussion by 134dd3v** [community/community#17666] [Pull Requests]
As an example in this PR https://github.com/RageTrade/core/pull/143, so many "`x` temporarily deployed to CI `y` minutes ago" msgs are annoying. As a suggestion I think, at least old ones could disappear.

**Comment by quezak** [comment]
I agree. We have so many such messages, that the PR page starts collapsing old events in the PR history, including unresolved comments and approvals, which are much more important to be visible IMO. Maybe it would be better to collapse just the github actions sections automatically, to keep us from lots of scrolling to view comments & review results?

**Reply by 403-html** [reply]
+1

**Reply by prgres** [reply]
+1

**Reply by mar-pan** [reply]
+1

**Reply by MichalJasiorowski** [reply]
+1

**Reply by 134dd3v** [reply]
btw @quezak we're using your OSS tools! Glad to brief e-meet here and thanks for all your contributions ❤️

**Reply by baracudda** [reply]
+1

**Reply by steveh-101** [reply]
+1

**Reply by crickettylerchan** [reply]
+1

**Reply by sichanyoo** [reply]
+1

**Reply by copiri-six** [reply]
`[User]` temporarily deployed to `[branch]` on January 1, 2024 12:15...
`[User]` temporarily deployed to `[branch]` on January 1, 2024 12:15...
`[User]` temporarily deployed to `[branch]` on January 1, 2024 12:15...

🤮 

Automated actions are great to have access to, but 99.999% of the time are never what is useful.  They should have their own tab so that things like conversations can be easily found and addressed.

**Reply by DonadoJuan** [reply]
+1

**Reply by 403-html** [reply]
Literally just something like this for Github Actions outputs would be sufficient (with removing them from the conversations tab)
![image](https://github.com/community/community/assets/57900160/a1c2e266-5036-470b-8ae3-83e60bd6544a)

**Reply by MarcinMiler** [reply]
+1

**Reply by kosta** [reply]
I just had to scroll through 500 messages of `@xxx xxx temporarily deployed to xxx` in an older PR (dependabot update that broke CI but a later update fixed itself). Is there anyone one can do to supress this message?

**Reply by kaisbaccour** [reply]
+1

**Reply by uncledru** [reply]
If you are setting the `environment:` in your job / workflow... Removing it will stop GitHub Actions from posting the messages.

**Reply by webbertakken** [reply]
> @uncledru: If you are setting the environment: in your job / workflow... Removing it will stop GitHub Actions from posting the messages.

While that's true, the environment is also used to provide environment specific `Environment variables` and `Environment secrets`, as well as show the status and links to deployed environments.

Practically speaking though, in case of `feature` environment (i.e. non-[`prod`|`beta`|`staging`] envs), we might get rid of the [environment](https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-what-your-workflow-does/deploying-to-a-specific-environment#using-an-environment-in-a-workflow) entirely and set its variables and secrets as `Repository variables` and `Repository secrets` (to be overridden by said `prod`, `beta` and `staging` envs).

---

I have just tested with `environment: 'My override test'` the following:

- setting MY_VAR in both `Repository variables` and `Environment variables` 
- setting MY_SECRET in both `Repository secrets` and `Environment secrets`

And and verified that in both cases the environment specific variable/secret takes precedence over the repository wide version. ✅

---

I realise this doesn't solve everyone's issues!
But at least it means that for feature branches this approach could work 🙂

**Comment by avbhandaru** [comment]
+1 any updates here?

**Comment by Splines** [comment]
This is such an amazing review experience. Why hide/fold the review comments?

https://github.com/user-attachments/assets/77976c5b-5eb6-4c16-84b6-e3d969d16ed9

**Reply by Splines** [reply]
As a workaround, you can use the [GitHub Pull Requests VSCode extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github) that doesn't show the annoying "Requested deployment" every single commit or hides the review messages behind 10 clicks. I fully agree with [this proposal](https://github.com/orgs/community/discussions/17666#discussioncomment-8838840).
